aten/src/ATen/native/mps/operations/RangeFactories.mm

```diff
@@ -13,6 +13,42 @@
 namespace at {
 namespace native {
 
+static const char* ARANGE_KERNEL = R"METAL(
+#include <metal_stdlib>
+
+using namespace metal;
+struct ARangeFloatParam {
+  float start;
+  float step;
+  uint length;
+};
+
+kernel void arange_float(constant ARangeFloatParam *params [[buffer(0)]],
+                         device float  *out [[buffer(1)]],
+                         uint offset[[thread_position_in_grid]]) {
+  if (offset >= params->length) {
+    return;
+  }
+  out[offset] = params->start + offset * params->step;
+}
+)METAL";
+
+static id<MTLComputePipelineState> compileMetalShader(id<MTLDevice> device) {
+ static id<MTLComputePipelineState> rc = nil;
```

Comment:
I think instead of this we should have a proper PSO cache which can be used for other operations.